packages/storage/src/components/UpdateUserPhotoModal/index.js

```diff
@@ -0,0 +1,194 @@
+
+    const pathProp = path || `users/${user ? user.uid : ''}/profilePhoto`;
+    const bucketProp = bucket || `${app.options.storageBucket}`;
+
+    const [photoURL, setPhotoURL] = useState(photo);
+    const [isOpenEdit, setIsOpenEdit] = useState(false);
+    const cropperRef = useRef(null);
+    const [loading, setLoading] = useState(false);
+
+    const handleOnClickEdit = () => {
+        onRequestClose();
+        setIsOpenEdit(true);
+    };
+
+    const onCrop = async () => {
+        const imageElement = cropperRef ? cropperRef.current : null;
+        const cropper = imageElement ? imageElement.cropper : null;
+        const storage = getStorage(app, bucketProp);
+        const photoRef = ref(storage, `${pathProp}/${nanoid()}`);
+        const blob = await fetch(cropper.getCroppedCanvas().toDataURL()).then((r) => r.blob());
+
+        const uploadTask = uploadBytesResumable(photoRef, blob);
+        uploadTask.on(
+            'state_changed',
+            (snapshot) => {
+                const progress = (snapshot.bytesTransferred / snapshot.totalBytes) * 100;
+                if (progress !== 100) {
+                    setLoading(true);
+                }
+            },
+            (error) => {
+                console.log(error);
+            },
+            async () => {
+                setIsOpenEdit(false);
+                setLoading(false);
+                const photoURL = await getDownloadURL(photoRef);
+                setPhotoURL(photoURL);
+                await updateProfile(user, {
+                    photoURL,
+                });
+                await reload();
+                handlePhotoUrl(photoURL);
+            },
+        );
+    };
+
+    const onClickRotate = () => {
+        const imageElement = cropperRef ? cropperRef.current : null;
+        const cropper = imageElement ? imageElement.cropper : null;
+        cropper.rotate(90);
+    };
+
+    const handleOnClickRemove = async () => {
+        const storage = getStorage(app);
+        const photoRef = ref(storage, photoURL);
+        await deleteObject(photoRef)
+            .then(() => {
+                updateProfile(user, {
+                    photoURL: '',
+                }).then(() => {
+                    setPhotoURL('');
+                    setIsOpenEdit(false);
+                    onRequestClose();
+                    reload().then(() => {
+                        handlePhotoUrl('');
+                    });
+                });
+            })
+            .catch((error) => {
+                console.log(error);
+            });
+    };
+
+    return (
+        <>
+            <StyledModal isOpen={isOpen} onRequestClose={onRequestClose}>
+                <EditRemoveDialog
+                    imageSrc={photo}
+                    avatarInitials={avatarInitials}
+                    handleOnClickEdit={handleOnClickEdit}
+                    handleImageSrc={setPhotoURL}
+                    handleOnClickRemove={handleOnClickRemove}
```

Comment:
normally for convention when you can to pass events to props you do this:
```
onClick={handleOnClick}
```
the word handle is used for  the handler but for the prop use onSomethign attaching the on always, like with native events